src/hotspot/share/utilities/concurrentHashTable.hpp

```diff
@@ -192,9 +192,11 @@ class ConcurrentHashTable : public CHeapObj<F> {
   };
 
   // Used as default functor when no functor supplied for some methods.
+  // Note it only accepts the VALUE, and does not define methods with
+  // non-void VALUE returns. Doing so would require defining the neutral
+  // value for VALUE.
```

Comment:
I don't think this comment is needed.  What's needed is a description of the DELETE_FN in the declaration of remove(), but that kind of template parameter requirements documentation is entirely missing from ConcurrentHashTable (and most other templates in HotSpot).